src/test/java/com/google/devtools/build/lib/packages/TargetUtilsTest.java

```diff
@@ -96,4 +98,142 @@ public void testExecutionInfo() throws Exception {
     execInfo = TargetUtils.getExecutionInfo(tag1b);
     assertThat(execInfo).containsExactly("local", "", "cpu:4", "");
   }
+
+  @Test
+  public void testExecutionInfo_withPrefixSupports() throws Exception {
+    scratch.file(
+        "tests/BUILD",
+        "sh_binary(name = 'with-prefix-supports', srcs=['sh.sh'], tags=['supports-workers', 'supports-whatever', 'my-tag'])"
+        );
+
+    Rule withSupportsPrefix = (Rule) getTarget("//tests:with-prefix-supports");
+
+    Map<String, String> execInfo = TargetUtils.getExecutionInfo(withSupportsPrefix);
+    assertThat(execInfo).containsExactly("supports-whatever", "", "supports-workers", "");
+  }
+
+  @Test
+  public void testExecutionInfo_withPrefixDisable() throws Exception {
+    scratch.file(
+        "tests/BUILD",
+        "sh_binary(name = 'with-prefix-disable', srcs=['sh.sh'], tags=['disable-local-prefetch', 'disable-something-else', 'another-tag'])"
+    );
+
+    Rule withDisablePrefix = (Rule) getTarget("//tests:with-prefix-disable");
+
+    Map<String, String> execInfo = TargetUtils.getExecutionInfo(withDisablePrefix);
+    assertThat(execInfo).containsExactly("disable-local-prefetch", "", "disable-something-else", "");
+  }
+
+  @Test
+  public void testExecutionInfo_withPrefixNo() throws Exception {
+    scratch.file(
+        "tests/BUILD",
+        "sh_binary(name = 'with-prefix-no', srcs=['sh.sh'], tags=['no-remote-imaginary-flag', 'no-sandbox', 'unknown'])"
+    );
+
+    Rule withNoPrefix = (Rule) getTarget("//tests:with-prefix-no");
+
+    Map<String, String> execInfo = TargetUtils.getExecutionInfo(withNoPrefix);
+    assertThat(execInfo).containsExactly("no-remote-imaginary-flag","", "no-sandbox", "");
+  }
+
+  @Test
+  public void testExecutionInfo_withPrefixRequires() throws Exception {
+    scratch.file(
+        "tests/BUILD",
+        "sh_binary(name = 'with-prefix-requires', srcs=['sh.sh'], tags=['requires-network', 'requires-sunlight', 'test-only'])"
+    );
+
+    Rule withRequiresPrefix = (Rule) getTarget("//tests:with-prefix-requires");
+
+    Map<String, String> execInfo = TargetUtils.getExecutionInfo(withRequiresPrefix);
+    assertThat(execInfo).containsExactly("requires-network", "", "requires-sunlight", "");
+  }
+
+  @Test
+  public void testExecutionInfo_withPrefixBlock() throws Exception {
+    scratch.file(
+        "tests/BUILD",
+        "sh_binary(name = 'with-prefix-block', srcs=['sh.sh'], tags=['block-some-feature', 'block-network', 'wrong-tag'])",
+        "sh_binary(name = 'with-prefix-cpu', srcs=['sh.sh'], tags=['cpu:123', 'wrong-tag'])",
+        "sh_binary(name = 'with-local-tag', srcs=['sh.sh'], tags=['local', 'some-tag'])"
```

Comment:
remove the above two lines. they are not used in the test.

Reply:
done, thanks :)